BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/RecentTransactionsActivity.java

```diff
@@ -0,0 +1,136 @@
+        Bank b = new BankFactory(v).create();
+
+        b.createSession()
+                .thenAccept(t -> Tools.runOnMainThread(() -> {
+                    Log.d("GUI", "Created session");
+                    Toast.makeText(getApplicationContext(),
+                            "Created session!",
+                            Toast.LENGTH_LONG).show();
+
+                    b.listTransactions().thenAccept(ts -> Tools.runOnMainThread(() -> {
+                        Toast.makeText(getApplicationContext(),
+                                "Got list of transactions!",
+                                Toast.LENGTH_LONG).show();
+                        Log.i("GUI", ts.toString());
+                        findViewById(R.id.loadingPanel).setVisibility(View.GONE);
+                        showRecentTransactions(ts);
+
+                    }));
+
+                }))
+                .exceptionally(e -> {
+                    final Throwable t = b.confirmException(e);
+
+                    Tools.runOnMainThread(() -> {
+                        Log.d("GUI", "Failed session " + t.getMessage());
+                        Toast.makeText(getApplicationContext(),
+                                "Failed to create session: " + t.getMessage(),
+                                Toast.LENGTH_LONG).show();
+                        Log.d("GUI", "Just failed");
+                        showRecentTransactions(null);
+                    });
+                    findViewById(R.id.loadingPanel).setVisibility(View.GONE);
+                    return null;
+                });
+    }
+
+    /*
+    Takes the list of recent transactions and outputs them to the UI
+     */
+    public void showRecentTransactions(List<? extends Transaction> transactionList) {
+        listDataHeader = new ArrayList<>();
+        listDataChild = new HashMap<>();
+
+        expListView = (ExpandableListView) findViewById(R.id.recentTransactionsList);
+        listAdapter = new ExpandableListAdapter(this, listDataHeader, listDataChild);
+        expListView.setAdapter(listAdapter);
+
+        if (transactionList == null) {
+            updateNoTransactionsDisplay("Could not retrieve transactions.");
+            return;
+        }
+        if (transactionList.size() == 0) {
+            updateNoTransactionsDisplay("No transactions have been made yet.");
+            return;
+        }
+
+        for (int i = 0; i < transactionList.size(); i++) {
+            listDataHeader.add(transactionList.get(i).getDescription());
+            List<String> details = new ArrayList<>();
+
+            if (transactionList.get(i).getValue() != null && transactionList.get(i).getCurrency() != null) {
+                details.add(transactionList.get(i).getValue().toString() + " " + transactionList.get(i).getCurrency().toString());
+            }
+            if (transactionList.get(i).getDirection() != null) {
+                details.add("" + transactionList.get(i).getDirection().toString());
+            }
```

Comment:
getDate() doesn't work correctly and throws an exception when called. The story was for the UI to display the information about a transaction, which it does for all available information.
I could remove the TODO and the commented code, but I think it's better to leave it as a reminder and uncomment when getDate() works.